igcollect/switch.py

```diff
@@ -328,6 +355,23 @@ def ports_stats(prefix, snmp, ports, model):
                 print(template.format(port_name, counter, data))
 
 
+def dom_stats(prefix, snmp, ports, model):
+    """
+    Print graphite-compatible Digital Optical Monitoring stats for each port
+    of the switch
+    """
+
+    for metric, oid in DOM_OIDS[model].items():
+        # SNMP is slow, so we want need to get current time for each counter.
```

Comment:
If it's so slow, then we should change this behaviour for all OIDs. Please check how it changes the total execution time and CPU load.

Reply:
Same comment as line 342. It was already there. 

I will check the impact on these metrics, nonetheless.

Reply:
This specific patch for MXL switches raised the execution time from 3,5 to 5,5 seconds. 
The user and system times kept pretty much the same, as most of the time is spent waiting for the SNMP responses.